### DISCUSSION TITLE
I made some utils for going back, forward between pages and copy text to clipboard.

### DISCUSSION THREAD

**Discussion by ZeroCool940711** [zauberzeug/nicegui#2575] [Ideas / Feature Requests]
Hi there, I recently needed some utilities like going back/forward to the previous/next page as well as copying some text to the clipboard and I couldn't find a way to do it with NiceGUI other than using `ui.run_javascript`, so, I decided to add a way to do just that but easier for reusing it in several projects, I have the change in a [different branch on my fork](https://github.com/zauberzeug/nicegui/compare/main...ZeroCool940711:nicegui:utils?expand=1) and was thinking that maybe I should submit a PR for it. Since I'm not sure how this kind of thing would be taken, I haven't made a PR and decided to instead open a discussion first to talk about this and any possible changes that can be done to it.

The initial implementation is pretty simple, I just used the browser `history.back()`, `history.fordward()` and `navigator.clipboard.writeText()` and called them with `ui.run_javascript()`,we then can just use `ui.back()`, `ui.fordward()` and `ui.copy_to_clipboard("text")` on any `on_click` event. The back and forward events seem to work just like in a normal browser, it also works in native mode and interesting enough it works with the back and forth buttons in a 7 button mouse, for reference in Flet you can go back by using BACKSPACE but not forward and the mouse back button doesn't work.

I think it could be expanded and improved further, maybe there are some issues with this way that I haven't seen but other have and there is a better way to do the same without that issue, so, let's talk about it. If @rodja and @falkoschindler like this little enhancement, I can make a PR for it.

**Comment by falkoschindler** [comment]
Thanks for the suggestion, @ZeroCool940711!

We thought about it and would propose to split it into two separate PRs:

1. Introducing a new `ui.navigate` object:
    ```py
    ui.navigate.to()
    ui.navigate.back()
    ui.navigate.forward()
    ```
    `ui.navigate.to()` basically resembles `ui.open`, which might get deprecated by the new API.

2. Introducing a new `ui.clipboard` object:
    ```py
    ui.clipboard.read()
    ui.clipboard.read_text()
    ui.clipboard.write()
    ui.clipboard.write_text()
    ```
    These are the methods from [JavaScript's Clipboard object](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard).

Both objects should be implemented in the "functions" directory, because they don't represent UI elements.

**Reply by ZeroCool940711** [reply]
I will look into implementing it the way you mentioned and will make two PRs for them, probably later today or tomorrow.

**Reply by ZeroCool940711** [reply]
@falkoschindler I've made PR #2593 and #2594 with a basic implementation of `ui.navigate` and `ui.clipboard`, the one for `ui.navigate` might be good as it is but I'm not so sure about the PR for `ui.clipboard` which is why I made it as a draft since it will probably require changes.
